cpp/test/sg/hdbscan_test.cu

```diff
@@ -444,5 +449,160 @@ INSTANTIATE_TEST_CASE_P(SoftClusteringTest,
+                                                                    params.n_row,
+                                                                    params.condensed_parents.size(),
+                                                                    condensed_parents.data(),
+                                                                    condensed_children.data(),
+                                                                    condensed_lambdas.data(),
+                                                                    condensed_sizes.data());
+
+    rmm::device_uvector<IdxT> label_map(params.n_row, handle.get_stream());
+
+    // intermediate outputs
+    rmm::device_uvector<T> stabilities(params.n_row, handle.get_stream());
+    rmm::device_uvector<T> probabilities(params.n_row, handle.get_stream());
+    rmm::device_uvector<IdxT> labels(params.n_row, handle.get_stream());
+
+    int n_selected_clusters =
+      ML::HDBSCAN::detail::Extract::extract_clusters(handle,
+                                                     condensed_tree,
+                                                     params.n_row,
+                                                     labels.data(),
+                                                     stabilities.data(),
+                                                     probabilities.data(),
+                                                     label_map.data(),
+                                                     params.cluster_selection_method,
+                                                     params.allow_single_cluster,
+                                                     0,
+                                                     params.cluster_selection_epsilon);
+
+    ML::HDBSCAN::Common::PredictionData<IdxT, T> pred_data(handle, params.n_row, params.n_col);
+
+    auto stream = handle.get_stream();
+    rmm::device_uvector<IdxT> mutual_reachability_indptr(params.n_row + 1, stream);
+    raft::sparse::COO<T, IdxT> mutual_reachability_coo(stream,
+                                                       params.min_samples * params.n_row * 2);
+
+    ML::HDBSCAN::detail::Reachability::mutual_reachability_graph(
+      handle,
+      data.data(),
+      (size_t)params.n_row,
+      (size_t)params.n_col,
+      raft::distance::DistanceType::L2SqrtExpanded,
+      params.min_samples,
+      (float)1.0,
+      mutual_reachability_indptr.data(),
+      pred_data.get_core_dists(),
+      mutual_reachability_coo);
+
+    ML::HDBSCAN::Common::build_prediction_data(
+      handle, condensed_tree, labels.data(), label_map.data(), n_selected_clusters, pred_data);
+
+    // outputs
+    rmm::device_uvector<IdxT> out_labels(params.n_points_to_predict, handle.get_stream());
+    rmm::device_uvector<T> out_probabilities(params.n_points_to_predict, handle.get_stream());
+
+    transformLabels(handle, labels.data(), label_map.data(), params.n_row);
+
+    ML::HDBSCAN::detail::Predict::approximate_predict(handle,
```

Comment:
I would suggest calling the public API function for this, since the approximate predict has one